website/blog/2023-01-12-version-071.md

```diff
@@ -0,0 +1,246 @@
+tags: [announcement, release]
+date: 2023-01-12
+---
+
+Today we’re releasing React Native version 0.71. This is a feature-packed release which includes TypeScript by default, Flexbox gap support, new web-inspired props, developer experience improvements, updates to the New React Native Architecture, and more!
+
+## TypeScript by default
+
+In this release, we’re investing in the TypeScript experience of React Native.
+
+Starting with 0.71, when you create a new React Native app via the React Native CLI you'll get a TypeScript app by default. The new project is already set up with a `tsconfig.json` so out of the box your IDE will help you write typed code right away.
+
+We’re also offering built-in TypeScript declarations directly from the `react-native` package. This means you won’t need `@types/react-native` any longer.
+
+Finally, our documentation has been updated to feature TypeScript for all examples.
+
+:::note
+
+After upgrading to React Native 0.71, we recommend removing `@types/react-native` from your package.json `devDependencies`.
+
+:::
+
+For more details on this change, including migration steps and how this affects Flow users, check out our previous post [_First-class Support for TypeScript_](https://reactnative.dev/blog/2023/01/03/typescript-first).
+
+## Simplifying layouts with Flexbox gap
+
+With React Native you can flexibly layout components on different screen sizes using flexbox. Browsers have supported the flexbox properties [_gap_](https://developer.mozilla.org/en-US/docs/Web/CSS/gap), [_rowGap_](https://developer.mozilla.org/en-US/docs/Web/CSS/row-gap), and [_columnGap_](https://developer.mozilla.org/en-US/docs/Web/CSS/column-gap), which allow you to specify the amount of space between all items in a flexbox.
+
+These properties have been long requested in React Native, and 0.71 adds initial support for gaps defined using pixel values. In future version, we will add support for more values, such as percentages.
+
+To see why this is useful, imagine trying to build a responsive layout with variably sized cards, 10px apart from each other, hugging the edges of the parent container. Trying to accomplish this layout with child margins can be tricky.
+
+The following shows a layout where we start by giving each child `margin: 10`:
+
+![Two diagrams. On the left it shows a skeleton of an app with three boxes that have margin around them cause the boxes to have margin around all sides. On the right, the same diagram is shown highlighted to demonstrate the margin on all sides.](/blog/assets/FlexboxGapBefore.png)
+
+Margins are applied uniformly to the edges of all children and don’t collapse under flexbox, giving us spacing at the exterior of the cards, and double the space on the interior compared to what we wanted. We can get around this by applying non-uniform margins, using negative margins on the parent, halving our intended spacing, etc, but it can be made much easier.
```

Comment:
```suggestion
Margins are applied uniformly to the edges of all children and don’t collapse under Flexbox, giving us spacing at the exterior of the cards, and double the space on the interior compared to what we wanted. We can get around this by applying non-uniform margins, using negative margins on the parent, halving our intended spacing, etc, but it can be made much easier.
```